### ISSUE TITLE
Issues with different servers

### ISSUE THREAD

**Issue by rthill** [librespeed/speedtest-cli#6]
I am having an issue running the librespeed-cli binary with multiple local json files containing each only one server entry to measure latency between my office and test/prod infrastructure.

On test infrastructure everything works with the following output in debug mode:
```
Using local JSON server list: TEST 
Selecting the fastest server based on ping
Failed to ping target host: socket: permission denied 
Will try TCP ping
Selected server: TEST [www.test.xxx.xxx] 
Failed when parsing get IP result: json: cannot unmarshal string into Go struct field GetIPResult.rawIspInfo of type defs.IPInfoResponse 
Received payload: {"processedString":"x.x.x.x","rawIspInfo":"UNKNOWN"}
You're testing from: x.x.x.x
Failed to ping target host: socket: permission denied
Will try TCP ping 
Download test is disabled 
Upload test is disabled
{"timestamp":"2020-04-01T16:33:47.921288388+02:00","server":{"name":"TEST","url":"https://www.test.xxx.xxx/librespeed/"},"client":{"ip":"","hostname":"","city":"","region":"","country":"","loc":"","org":"","postal":"","timezone":""},"bytes_sent":0,"bytes_received":0,"ping":10.272727272727273,"jitter":0.53,"upload":0,"download":0,"share":""}
```

On production server I get only the following output:
```
Using local JSON server list: PROD
Selecting the fastest server based on ping
Server PROD (www.xxx.xxx) doesn't seem to be up, skipping 
No server is currently available, please try again later.
```

Both servers do not respond to ICMP ping requests and I have no clue why the test server reports `Failed to ping target host: socket: permission denied` and then continues with a TCP ping? Is there a possibility to always switch or fallback to a "TCP ping" when no ICMP response received?

Thanks for your help,
Raoul

**Comment by maddie** [comment]
TCP pings are used as a fallback in the case of ICMP pings are failed, as indicated in the debug log you've posted:

```
Failed to ping target host: socket: permission denied
Will try TCP ping  <<<<
```

I presume you're using Linux? Check the following link for the `Failed to ping target host: socket: permission denied` error:

https://github.com/sparrc/go-ping#note-on-linux-support

If TCP ping fails too, then please check that the `pingURL` endpoint is working correctly.

**Comment by rthill** [comment]
Yes I am on Linux, but as I don't have root access to any of the servers and firewalls are dropping ICMP packets the go-ping modifications cannot be applied.
The unmarshal failed may be ignored as this is due the modifications I made to `getIP.php`, essentially I removed everything in relation with external lookups to ipinfo service.

I've checked further and the only difference I could find was a trailing new line at the end of empty.php which is used as `pingURL`. Removing the trailling new line did solve my issue, but I dont have an explanation for this as there should be no difference between
```
?>

```
and
```
?>
```

I did also switch to librespeed-cli v1.0.6, thanks for returning json output now on STDOUT!
